src/inference/src/threading/ie_istreams_executor.cpp

```diff
@@ -241,6 +323,64 @@ Parameter IStreamsExecutor::Config::GetConfig(const std::string& key) const {
     return {};
 }
 
+int IStreamsExecutor::Config::LimitHybridStreams(const int num_streams, const int num_threads) {
+    const auto core_types = custom::info::core_types();
+    const auto num_big_cores_phys = getNumberOfCPUCores(true);
+    const auto num_small_cores =
+        custom::info::default_concurrency(custom::task_arena::constraints{}.set_core_type(core_types.front()));
+    const auto base_streams_total = num_big_cores_phys + num_small_cores / 2;
+
+    // limit streams to reasonable range
+    int streams = num_streams > base_streams_total ? base_streams_total : std::max(1, num_streams);
+    if (num_threads > 0 && streams > num_threads) {
+        streams = num_threads;
+    }
+    return streams;
+}
+
+void IStreamsExecutor::Config::UpdateHybridCustomThreads(Config& config,
+                                                         const int num_big_cores_phys,
+                                                         const int num_small_cores) {
+    const auto base_streams_total = num_big_cores_phys + num_small_cores / 2;
+    const auto num_cores_phys = num_big_cores_phys + num_small_cores;
+
+    config._small_core_offset = num_big_cores_phys * 2;
+    if (config._threads) {
+        // limit threads to reasonable range
+        auto threads = config._threads > num_cores_phys ? num_cores_phys : config._threads;
+        // all threads are placed on big cores
+        if (threads <= num_big_cores_phys) {
+            config._big_core_streams = config._streams;
+            config._threads_per_stream_big = std::max(1, threads / config._streams);
+        } else {
+            // big cores first, then small cores
+            config._big_core_streams =
+                config._streams * num_big_cores_phys % base_streams_total == 0
+                    ? config._streams * num_big_cores_phys / base_streams_total
+                    : (config._streams * num_big_cores_phys + base_streams_total) / base_streams_total;
+            config._small_core_streams = config._streams - config._big_core_streams;
+            config._threads_per_stream_big = std::max(1, num_big_cores_phys / config._big_core_streams);
+            if (config._small_core_streams > 0) {
+                const int remain_threads = threads - config._threads_per_stream_big * config._big_core_streams;
+                config._threads_per_stream_small =
+                    std::min(remain_threads / config._small_core_streams, config._threads_per_stream_big * 2);
```

Comment:
std::min or std::max?

Reply:
It should be std::min, if use std::max, it exceeds the number of small core in special condition.